### ISSUE TITLE
Zip Codes by City or State

### ISSUE THREAD

**Issue by quequegg** [Discordanian/mmatch#133]
I believe the zip code basis for identifying an organization is properly designed but it is a hassle in that the UI does not allow for a number of zips to be added in a batch. It is relatively easy to fix this with DML but it does not echo back to the user except as an indecipherable list of zips.

A couple of points:
a) This is a low volume activity and we should not over-invest
b) It is a gate-keeper; it makes it hard to get organizations over the initial set up so it is worth addressing.
c) Whatever is done ought to be saved so that we can echo back their selection even after logging out.

Any of these ideas could be implemented by themselves or with some additional thought, in combination:
1) Mirror the way the user selects zips; that is we could enter a single zip and radius and insert the zips that meet that criteria.
2) Let each installation create a list of municipalities in their region. (AZ would be the state; STL would be just those in the standard metropolitan area).  Put that list in a drop down from which the user could make a multi-select.  For an existing organization, when the drop down populates, the user's prior selections are checked for them.  When they submit, the list of zips are purged and replaced by the zips that match the drop down.
2a) Select all option.  Checking this would include all municipalities in the list above.

**Comment by alandgri** [comment]
I agree the interface needs work and I have already started thinking along those lines. The direction that I was going was allowing the user to add by city by typing in and searching by city name. The way I was thinking is the user types a few letters from the city name, they get a list of matching cities. When they click on one, it adds the zip code (or codes) for that city to the list.

A couple things you said I didn't get: 
1 - "it does not echo back to the user except as an indecipherable list of zips".
2 - "it ought to be saved so that we can echo back their selection"
Are you referring to the way the zip interaction piece looks on a phone or an iPad? On a computer it displays the list that is selected.

I'm not challenging the notion that it needs to be fixed, I just want to make sure I am understanding this particular issue.

**Comment by quequegg** [comment]
no worries ... I just mean that a list of zip codes is hard to make sense of.
And #2 for example, if I select a city instead of a zip then when I come back to the site I should see that city selected, not just the list of zips.  We would need a new table for this.

**Comment by alandgri** [comment]
We talked about this in person, and I think we determined that (for now) the only change necessary is to add the ability to add and subtract by city in the user interface. This is an inherently local movement, and therefore, there aren't many valid use cases for having a huge list of hundreds of zip codes. Even if you are a statewide organization, you can be found in searches by adding a few key cities.

**Comment by quequegg** [comment]
I agree. Also:
- Cases
1) I add a city causing zips in that city to add to zip list. Then I remove
a zip from within that city. I would expect to have all zips in the city
minus the zip I dropped.
2) I add a city where many zips are already included. I expect the zip code
list be filled in with those zips.

Other (do you prefer I break these out into separate issues?)
- Sort zip list by city so I can more easily see the effect of adding
cities.  (Bonus ... a sort by button.)
- Be careful of duplicates; with the unique index we need to watch for db
error messages.
- Consider putting a "?" button with the opportunity for some short
explanatory text.
- Zips not showing up in Safari on iPad.
